modules/core/03-connection/keeper/verify.go

```diff
@@ -195,9 +205,15 @@ func (k Keeper) VerifyPacketAcknowledgement(
 		return sdkerrors.Wrapf(clienttypes.ErrClientNotActive, "client (%s) status is %s", clientID, status)
 	}
 
+	// calculate minimum block delay by dividing time delay period
+	// by the maximum expected time per block. Round up the block delay.
+	timeDelay := connection.GetDelayPeriod()
+	maxTimePerBlock := k.GetMaxTimePerBlock(ctx)
+	blockDelay := uint64(math.Ceil(float64(timeDelay) / float64(maxTimePerBlock)))
```

Comment:
this enforces block delays even for time delays of 0 (since it rounds up). Is there documentation for why you chose to round up?

For the case where time delay > 0 and < maxTimePerBlock, the packet cannot be processed until the next block anyways, regardless of block delay since the time delay is > 0. So the cases we only care about are time delay == 0 and when timeDelay > MaxTimePerBlock and timeDelay % MaxTimePerBlock != 0

Lets say MaxTimePerBlock is 10s and my time delay is 35s. Rounding up puts us at 4 blocks and rounding down puts us at 3 blocks. In the usual worst case scenario, 3 blocks would be produced in 30 seconds, thus even if the block delay was 3, we would still need to wait for the 4th block. So I guess it doesn't really matter whether we round up or down for this case

For timeDelay == 0, we need to add an if statement if we round up, but no if statement is needed if we round down. This seems like the only reason to influence which way to round

Thoughts?

Reply:
> this enforces block delays even for time delays of 0 (since it rounds up). Is there documentation for why you chose to round up?

Not true: https://play.golang.org/p/5B4Sdhvs4PU

> Lets say MaxTimePerBlock is 10s and my time delay is 35s. Rounding up puts us at 4 blocks and rounding down puts us at 3 blocks. In the usual worst case scenario, 3 blocks would be produced in 30 seconds, thus even if the block delay was 3, we would still need to wait for the 4th block. So I guess it doesn't really matter whether we round up or down for this case

Rounding up is the safer approach. If the expected blocks are 3.3 blocks, its safer to wait 4 blocks because that is still fitting with our invariant that we wait **at least** "3.3" blocks. If we round down, it's not a huge deal but it's also breaking the security assumption. Since block delay itself is a conservative decision may as well stick to that philosophy. Also in cases where the calculated delay is 0.3 blocks, we want to at least wait one block. If we round down, we need to create a special case for this.